core/kernel/thread.c

```diff
@@ -31,17 +32,17 @@ struct thread_core_local thread_core_local[CFG_TEE_CORE_NB_CORE] __nex_bss;
  */
 
 #ifdef CFG_WITH_STACK_CANARIES
-#define START_CANARY_VALUE	0xdededede
-#define END_CANARY_VALUE	0xabababab
+static uintptr_t start_canary_value = 0xdedede00;
+static uintptr_t end_canary_value = 0xababab00;
 #define GET_START_CANARY(name, stack_num) name[stack_num][0]
 #define GET_END_CANARY(name, stack_num) \
-	name[stack_num][sizeof(name[stack_num]) / sizeof(uint32_t) - 1]
+	name[stack_num][sizeof(name[stack_num]) / sizeof(uintptr_t) - 1]
 #endif
 
 #define DECLARE_STACK(name, num_stacks, stack_size, linkage) \
-linkage uint32_t name[num_stacks] \
+linkage uintptr_t name[num_stacks] \
```

Comment:
We've used `uint32_t` on 64-bit platforms before this. If you want to change this, please do that in a separate PR to keep the focus on what's essential.

Reply:
Got it. I'll revert these changes